frontend/apps/filemanager/filemanager.lua

```diff
@@ -1279,6 +1285,19 @@ function FileManager:showFiles(path, focused_file)
     UIManager:show(file_manager)
 end
 
+function FileManager:openTextViewer(file_path)
+    local file = io.open(file_path, "rb")
+    if not file then return end
+    local file_content = file:read("*all")
```

Comment:
It might be an idea to add a size check warning? We'd need to experiment a little, but I mean for when it would result in a loading time of more than about 2-3 seconds on a 5 y/o device or more than 100 MB RAM used.

Without testing it I have no idea when such conditions would be met, but my guess would be somewhere in between 1 MB to 20 MB. I might perform some quick tests later if I remember.

Reply:
In Text editor warning threshold is 200 KB.
On my Kindle Voyage text viewer opening of the file of 1 MB takes 4-5 sec.

Reply:
In that case I'd suggest 400 or 500 kB. The text editor has a bit more to do presumably.